otherlibs/systhreads/st_win32.h

```diff
@@ -134,6 +139,19 @@ Caml_inline void st_tls_set(st_tlskey k, void * v)
   TlsSetValue(k, v);
 }
 
+/* OS-specific handling of the OCaml thread ID (must be called with the runtime
+   lock). */
+Caml_inline void st_thread_set_id(intnat id)
+{
+  st_tls_set(st_thread_id_key, (void *)id);
```

Comment:
What about an assertion `id != 0` ?  Just to enforce the requirement that no thread has ID 0.

Reply:
Done (and also in `st_current_thread_id`)